Mage.Sets/src/mage/cards/t/TheFourthDoctor.java

```diff
@@ -0,0 +1,243 @@
+package mage.cards.t;
+
+import java.util.*;
+import mage.MageIdentifier;
+import mage.MageInt;
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.SpellAbility;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.hint.Hint;
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.abilities.effects.common.continuous.LookAtTopCardOfLibraryAnyTimeEffect;
+import mage.constants.*;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.FoodToken;
+import mage.players.Player;
+import mage.watchers.Watcher;
+
+
+/**
+ * @author padfoothelix
+ */
+public final class TheFourthDoctor extends CardImpl {
+
+    public TheFourthDoctor(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{G}{U}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.TIME_LORD);
+        this.subtype.add(SubType.DOCTOR);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // You may look at the top card of your library any time.
+        this.addAbility(new SimpleStaticAbility(new LookAtTopCardOfLibraryAnyTimeEffect()));
+	
+        // adapted from Serra Paragon
+	// Would You Like A...? -- Once each turn, you may play a historic land or cast a historic spell from the top of your library. When you do, create a Food token.
+        this.addAbility(
+                new SimpleStaticAbility(
+                        new TheFourthDoctorPlayFromTopEffect())
+                        .setIdentifier(MageIdentifier.TheFourthDoctorWatcher)
+                        .addHint(TheFourthDoctorWatcher.getHint())
+                        .withFlavorWord("Would You Like A…?"),
+                new TheFourthDoctorWatcher()
+        );
+        this.addAbility(new TheFourthDoctorTriggeredAbility());
+    }
+
+    private TheFourthDoctor(final TheFourthDoctor card) {
+        super(card);
+    }
+
+    @Override
+    public TheFourthDoctor copy() {
+        return new TheFourthDoctor(this);
+    }
+}
+
+// adapted from PlayFromTopOfLibraryEffect
```

Comment:
Try to extend instead copy-paste. Call super.applies. If it true (card is playable) then add additional checks on historic and once per turn.